common/djangoapps/student/cookies.py

```diff
@@ -0,0 +1,141 @@
+        user (User): The currently logged in user.
+
+    Returns:
+        HttpResponse
+
+    """
+    if request.session.get_expire_at_browser_close():
+        max_age = None
+        expires = None
+    else:
+        max_age = request.session.get_expiry_age()
+        expires_time = time.time() + max_age
+        expires = cookie_date(expires_time)
+
+    cookie_settings = {
+        'max_age': max_age,
+        'expires': expires,
+        'domain': settings.SESSION_COOKIE_DOMAIN,
+        'path': '/',
+        'httponly': None,
+    }
+
+    # Backwards compatibility: set the cookie indicating that the user
+    # is logged in.  This is just a boolean value, so it's not very useful.
+    # In the future, we should be able to replace this with the "user info"
+    # cookie set below.
+    response.set_cookie(settings.EDXMKTG_LOGGED_IN_COOKIE_NAME, 'true', secure=None, **cookie_settings)
+
+    # Set a cookie with user info.  This can be used by external sites
+    # to customize content based on user information.  Currently,
+    # we include information that's used to customize the "account"
+    # links in the header of subdomain sites (such as the marketing site).
+    header_urls = {'logout': reverse('logout')}
+
+    # Unfortunately, this app is currently used by both the LMS and Studio login pages.
+    # If we're in Studio, we won't be able to reverse the account/profile URLs.
+    # To handle this, we don't add the URLs if we can't reverse them.
+    # External sites will need to have fallback mechanisms to handle this case
+    # (most likely just hiding the links).
+    try:
+        header_urls['account_settings'] = reverse('account_settings')
+        header_urls['learner_profile'] = reverse('learner_profile', kwargs={'username': user.username})
+    except NoReverseMatch:
+        pass
+
+    # Convert relative URL paths to absolute URIs
+    for url_name, url_path in header_urls.iteritems():
+        header_urls[url_name] = request.build_absolute_uri(url_path)
+
+    user_info = {
+        'version': settings.EDXMKTG_USER_INFO_COOKIE_VERSION,
+        'username': user.username,
+        'email': user.email,
+        'header_urls': header_urls,
+    }
+
+    # In production, TLS should be enabled so that this cookie is encrypted
+    # when we send it.  We also need to set "secure" to True so that the browser
+    # will transmit it only over secure connections.
+    #
+    # In non-production environments (acceptance tests, devstack, and sandboxes),
+    # we still want to set this cookie.  However, we do NOT want to set it to "secure"
+    # because the browser won't send it back to us.  This can cause an infinite redirect
+    # loop in the third-party auth flow, which calls `is_logged_in_cookie_set` to determine
+    # whether it needs to set the cookie or continue to the next pipeline stage.
+    user_info_cookie_is_secure = request.is_secure()
```

Comment:
Note to reviewers: this fixes an issue that was identified by the acceptance tests.  I prefer this approach to introducing another setting to control whether the cookie is secure or not, but I'm interested to hear other opinions.